Podcast/SeriesDetailViewController.swift

```diff
@@ -20,76 +21,139 @@ class SeriesDetailViewController: UIViewController, SeriesDetailHeaderViewDelega
     
     var seriesHeaderView: SeriesDetailHeaderView!
     var epsiodeTableView: UITableView!
+    var loadingAnimation: NVActivityIndicatorView!
+    
+    var series: Series?
+    let pageSize = 20
+    var offset = 0
+    var continueInfiniteScroll = true
     
-    var series: Series!
-
     override func viewDidLoad() {
         super.viewDidLoad()
-        fetchSeries()
-        createSubviews()
-        updateViewsWithSeries(series: series)        
-    }
-    
-    func createSubviews() {
-        let seriesHeaderViewframe = CGRect(x: 0, y:0, width: view.frame.width, height: seriesHeaderHeight)
-        seriesHeaderView = SeriesDetailHeaderView(frame: seriesHeaderViewframe)
+        guard let appDelegate = UIApplication.shared.delegate as? AppDelegate else { return }
+        seriesHeaderView = SeriesDetailHeaderView(frame: CGRect(x: 0, y: 0, width: view.frame.width, height: seriesHeaderHeight))
         seriesHeaderView.delegate = self
+        seriesHeaderView.isHidden = true
         
-        guard let appDelegate = UIApplication.shared.delegate as? AppDelegate else { return }
-        let tableViewframe = CGRect(x: 0, y: 0, width: view.frame.width, height: view.frame.height - appDelegate.tabBarController.tabBarHeight)
-        epsiodeTableView = UITableView(frame: tableViewframe, style: .plain)
+        var seriesHeaderViewY: CGFloat = 0
+        if let height = navigationController?.navigationBar.frame.maxY  {
+            seriesHeaderViewY = height
+        }
+        epsiodeTableView = UITableView(frame:  CGRect(x: 0, y: seriesHeaderViewY, width: view.frame.width, height: view.frame.height - seriesHeaderViewY))
         epsiodeTableView.delegate = self
         epsiodeTableView.dataSource = self
         epsiodeTableView.tableHeaderView = seriesHeaderView
         epsiodeTableView.showsVerticalScrollIndicator = false
+        epsiodeTableView.separatorStyle = .none
+        epsiodeTableView.contentInset = UIEdgeInsetsMake(0, 0, appDelegate.tabBarController.tabBarHeight, 0)
+        epsiodeTableView.addInfiniteScroll { (tableView) -> Void in
+            self.fetchEpisodes()
+        }
+        //tells the infinite scroll when to stop
+        epsiodeTableView.setShouldShowInfiniteScrollHandler { _ -> Bool in
+            return self.continueInfiniteScroll
+        }
         epsiodeTableView.register(EpisodeTableViewCell.self, forCellReuseIdentifier: "EpisodeTableViewCellIdentifier")
-        
         view.addSubview(epsiodeTableView)
+    
+        epsiodeTableView.infiniteScrollIndicatorView = createLoadingAnimationView()
+        
+        if let series = self.series {
```

Comment:
Ok so in search we have full series we are passing in. Since we already get full series from search we don't need another network request before the view is presented but only need one to get the episodes. However in subscriptions since our data is denormalized on backend we only get part of a series so I make a network request to get the full series and show the loading indicator while I do that